CONTRIBUTING.md

```diff
@@ -1,30 +1,251 @@
-   [PEP8](https://www.python.org/dev/peps/pep-0008) style guidelines.
+There are many ways of contributing: from catching a typo to coming up with a way
+of improving performance or accessibility; you can open an issue, or you can prepare
+a patch. In any case, read the contribution guidelines for opening new issues and
+submitting pull requests.
 
-   You can run
-   ```shell script
-   tox -elint
-   ```
-   from [`client/`, `manager/`] folders for lint conformance checks.
+
+## Before you start
+
+Contributing to Quantum Serverless, assumes you have some level
+of [Git](https://git-scm.com) knowledge. For external contributors, a basic understanding
+of repositories, remotes, branches and commits is needed. For core contributors, you
+should know about resolving conflicts and rebasing too.
+
+There are tons of useful resources about Git [out there](https://try.github.io/).
+
+
+## Opening issues
+
+You can [open 4 types of issues](https://github.com/Qiskit-Extensions/quantum-serverless/issues/new/choose):
+
+* Bug reports: for reporting a misfunction. Provide steps to reproduce and expected behaviour.
+* Enhancement request: to suggest improvements to the current code.
+* Feature request: if you have a new use case or feature that we are not supporting.
+* Security vulnerability: in case you find a vulnerability in the project.
+
+Core contributors classify the tasks according to its nature and prioritize them
+from sprint to sprint. Types are not mutually exclusive and can change over time
+if needed.
+
+
+## Contributing code
+
+
+### Deciding what to work on
+
+To give our collaborators an idea of where the team needs help, we use the
+[help wanted](https://github.com/Qiskit-Extensions/quantum-serverless/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
+label – this is appropriate for all contributors. In addition, for those who are relatively new to the open-source
+workflow or our codebase, feel free to view issues tagged with the
+[good first issue](https://github.com/Qiskit-Extensions/quantum-serverless/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+label.
+
+
+### Setup
+
+So you decided to get your hands dirty and start working on a patch? Then you
+need to know that the project follows the
+[Forking Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow)
+with [Feature Branches](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow).
+
+The above means we expect you to fork the project on your own GitHub account and make your `main` branch to
+track this repository. A typical Git setup after
+[forking the project](https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/fork-a-repo) is:
+
+```sh
+# After forking the repository in GitHub
+git clone https://github.com/<your_username>/quantum-serverless.git
+cd quantum-serverless
+git remote add upstream https://github.com/Qiskit-Extensions/quantum-serverless.git
```

Comment:
```suggestion
git remote add upstream https://github.com/Qiskit-Extensions/quantum-serverless.git
git remote set-url --push upstream no_push
```

(to prevent anyone with write access from accidentally pushing directly to upstream main)